src/pages/iou/steps/MoneyRequestAmountPage.js

```diff
@@ -74,24 +106,63 @@ class MoneyRequestAmountPage extends React.Component {
     }
 
     componentDidMount() {
-        this.focusTextInput();
+        if (this.isEditing) {
+            const moneyRequestId = `${this.iouType}${this.reportID}`;
+            const shouldReset = this.props.iou.id !== moneyRequestId;
+            if (shouldReset) {
+                IOU.resetMoneyRequestInfo(moneyRequestId);
+            }
+
+            if (_.isEmpty(this.props.iou.participants) || this.props.iou.amount === 0 || shouldReset) {
+                Navigation.goBack(ROUTES.getMoneyRequestRoute(this.iouType, this.reportID), true);
+                return;
+            }
+        }
 
         // Focus automatically after navigating back from currency selector
         this.unsubscribeNavFocus = this.props.navigation.addListener('focus', () => {
             this.focusTextInput();
-            this.getCurrencyFromRouteParams();
         });
     }
 
     componentDidUpdate(prevProps) {
-        if (prevProps.iou.selectedCurrencyCode === this.props.iou.selectedCurrencyCode) {
-            return;
+        if (this.isEditing) {
+            // ID in Onyx could change by initiating a new request in a separate browser tab or completing a request
+            if (_.isEmpty(this.props.iou.participants) || this.props.iou.amount === 0 || prevProps.iou.id !== this.props.iou.id) {
+                // The ID is cleared on completing a request. In that case, we will do nothing.
+                if (this.props.iou.id) {
+                    Navigation.goBack(ROUTES.getMoneyRequestRoute(this.iouType, this.reportID), true);
+                }
+                return;
+            }
+        }
+
+        const prevCurrencyParam = lodashGet(prevProps.route.params, 'currency', '');
+        const currencyParam = lodashGet(this.props.route.params, 'currency', '');
+        if (currencyParam !== '' && prevCurrencyParam !== currencyParam) {
+            this.setState({selectedCurrencyCode: currencyParam});
+        }
+
+        if (prevProps.iou.currency !== this.props.iou.currency) {
+            this.setState({selectedCurrencyCode: this.props.iou.currency});
         }
 
-        this.setState({selectedCurrencyCode: this.props.iou.selectedCurrencyCode});
+        if (prevProps.iou.amount !== this.props.iou.amount) {
```

Comment:
This has caused a small regression in https://github.com/Expensify/App/issues/22361
The dot wasn't removed from the end of amount when navigating back to the screen (in an edge case where amount is the same, more details [here](https://github.com/Expensify/App/issues/22361))
We resolved this by moving CurrencyUtils.convertToWholeUnit logic into a separate method and calling it in `useFocusEffect` (when page is focused)